File: herminedb-driver/src/main/java/org/hermine/db/driver/HermineDataSourceFactory.java
/*
 * Copyright 2018 HermineDB's author : Frédéric Montariol,
 * and explicitly declared author of this file if provided.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.hermine.db.driver;

import jdk.incubator.sql2.DataSource;
import jdk.incubator.sql2.DataSourceFactory;
import org.hermine.internal.db.driver.HermineDataSourceBuilderImpl;

public class HermineDataSourceFactory implements DataSourceFactory {

    @Override
    public DataSource.Builder builder() {
        return new HermineDataSourceBuilderImpl();
    }
}

File: herminedb-driver/src/main/kotlin/org/hermine/db/driver/HermineDataSourceFactory.kt
/*
 * Copyright 2018 HermineDB's author : Frédéric Montariol. Use of this source code is governed by the Apache 2.0 license.
 */

package org.hermine.db.driver

import jdk.incubator.sql2.DataSource
import jdk.incubator.sql2.DataSourceFactory
import org.hermine.internal.db.driver.DataSourceBuilderImpl

class HermineDataSourceFactory : DataSourceFactory {
    override fun builder(): DataSource.Builder = DataSourceBuilderImpl()
}
